### DISCUSSION TITLE
Submit Data in a Custom JSON format

### DISCUSSION THREAD

**Discussion by nvijayap** [vueform/vueform#106] [Questions] [answered]
Hi,

How to gather the data from all fields and submit it as a consolidated json (in a custom json format)? The reason I am inquiring on this is - I have the need to use nested json objects and nested json arrays in the custom json to be submitted. How can I generate the custom json to be submitted with nested json objects/arrays?

Thanks

**Comment by adamberecz** [comment, marked as answer]
Hi, you can use `:form-data` to transform the form data before submitting:
https://vueform.com/reference/vueform#option-form-data

Example:
```html
<template>
  <Vueform :form-data="formatData" ... />
</template>

<script setup>
const formatData = (form$) => {
  let data = form$.requestData // this is the full form data in JSON

  data = {
    nested: data
  } // transform the data

  return form$.convertFormData(data) // returns a FormData object
}
</script>
```

Alternatively you can access `requestData` / `data` directly:
```html
<template>
  <Vueform ref="form$" ... />
</template>

<script setup>
import { ref, onMounted } from 'vue'

const form$ = ref(null)

onMounted(() => {
  const data = form$.value.requestData // this is the full form data in JSON
})
</script>
```

... or when using a custom endpoint:
```html
<template>
  <Vueform :endpoint="customEndpoint" ... />
</template>

<script setup>
const customEndpoint = (FormData, form$) => {
  let data = form$.requestData // this is the full form data in JSON

  // handling form submission manually
}
</script>
```

You can read more about `requestData` vs `data` here:
https://vueform.com/docs/handling-form-data#data-vs-request-data
